Miscellaneous/Diameter-Division/akshat.cpp

```diff
@@ -0,0 +1,98 @@
+#include<bits/stdc++.h>
+using namespace std;
+#ifndef ONLINE_JUDGE
+#ifdef __linux__
+#include "/media/gakshat468/New Volume/CP/algo/debug.h"
+#elif _WIN32
+#include "D:/CP/algo/debug.h"
+#endif
+#else
+#define debug(...) 468
+#endif
+#define MOD 1000000007
+#define nl <<'\n'
+#define sp <<" "<<
+#define spe <<" "
+#define display_vec(a) for(auto i:a)cout<<i<<" "; cout <<"\n";
+#define fast_io ios_base::sync_with_stdio(false); cin.tie(NULL)
+#define ll long long
+bool solve()
+{
+    int n;
+    cin >> n;
+    vector<int> adj[n + 1];
+    for (int i = 0;i < n - 1;i++) {
+        int u, v;
+        cin >> u >> v;
+        adj[u].push_back(v);
+        adj[v].push_back(u);
+    }
+    int flim = sqrt(n * log2(n)) + 2;
```

Comment:
it would be better if this is done dynamically, something like d<=(kmin*overhead) than using statically calculated value